sdk/tables/azure-tables/review/tables.api.review.md

```diff
@@ -0,0 +1,695 @@
+  createEntity<T>(
+    tableName: string,
+    entity: Entity<T>,
+    options?: CreateEntityOptions
+  ): Promise<CreateEntityResponse>;
+  createTable(tableName: string, options?: CreateTableOptions): Promise<CreateTableResponse>;
+  deleteEntity(
+    tableName: string,
+    partitionKey: string,
+    rowKey: string,
+    ifMatch: string,
+    options?: DeleteEntityOptions
+  ): Promise<DeleteEntityResponse>;
+  deleteTable(tableName: string, options?: DeleteTableOptions): Promise<DeleteTableResponse>;
+  static fromConnectionString(
+    connectionString: string,
+    options?: TableServiceClientOptions
+  ): TableServiceClient;
+  getAccessPolicy(
+    tableName: string,
+    options?: GetAccessPolicyOptions
+  ): Promise<GetAccessPolicyResponse>;
+  getEntity<T extends object>(
+    tableName: string,
+    partitionKey: string,
+    rowKey: string,
+    options?: GetEntityOptions
+  ): Promise<GetEntityResponse<T>>;
+  getProperties(options?: GetPropertiesOptions): Promise<GetPropertiesResponse>;
+  getStatistics(options?: GetStatisticsOptions): Promise<GetStatisticsResponse>;
+  listEntities<T extends object>(
+    tableName: string,
+    query?: QueryOptions,
+    options?: ListEntitiesOptions
+  ): Promise<ListEntitiesResponse<T>>;
+  listTables(query?: QueryOptions, options?: ListTablesOptions): Promise<ListTablesResponse>;
+  setAccessPolicy(
+    tableName: string,
+    acl?: SignedIdentifier[],
+    options?: SetAccessPolicyOptions
+  ): Promise<SetAccessPolicyResponse>;
+  setProperties(
+    properties: ServiceProperties,
+    options?: SetPropertiesOptions
+  ): Promise<SetPropertiesResponse>;
+  updateEntity<T>(
+    tableName: string,
+    entity: Entity<T>,
+    mode?: UpdateMode,
+    eTag?: string,
+    options?: UpdateEntityOptions
+  ): Promise<UpdateEntityResponse>;
+  upsertEntity<T>(
+    tableName: string,
+    entity: Entity<T>,
+    mode?: UpdateMode,
+    options?: UpsertEntityOptions
+  ): Promise<MergeEntityResponse>;
+}
+
+// @public
+export interface TableServiceClientOptions extends coreHttp.ServiceClientOptions {
+  endpoint?: string;
+  version?: string;
+}
+
+// @public
+export interface TableServiceStats {
+  geoReplication?: GeoReplication;
+}
+
+// @public
+export interface TableSetAccessPolicyHeaders {
+  clientRequestId?: string;
+  date?: Date;
+  requestId?: string;
+  version?: string;
+}
+
+// @public
+export interface TableSetAccessPolicyOptionalParams extends coreHttp.OperationOptions {
+  requestId?: string;
+  tableAcl?: SignedIdentifier[];
+  timeout?: number;
+}
+
+// @public
+export interface TableUpdateEntityHeaders {
+  clientRequestId?: string;
+  date?: Date;
+  eTag?: string;
+  requestId?: string;
+  version?: string;
+}
+
+// @public
+export interface TableUpdateEntityOptionalParams extends coreHttp.OperationOptions {
+  ifMatch?: string;
+  queryOptions?: GeneratedQueryOptions;
+  requestId?: string;
+  tableEntityProperties?: {
+    [propertyName: string]: any;
```

Comment:
If possible we should override uses of `[propertyName: string]: any]` with generic types